### PR TITLE
vlan: update vagrant config template for vlan enhancement

### PR DESCRIPTION
**Pull request by tjjh89017**
update for https://github.com/harvester/harvester-installer/pull/278

Signed-off-by: Date Huang <date.huang@suse.com>

### CODE REVIEW

File: vagrant-pxe-harvester/ansible/roles/harvester/templates/config-join.yaml.j2 [hunk 1]
```diff
@@ -15,15 +15,10 @@ os:
...
-      interfaces:
-      - name: {{ settings['harvester_network_config']['cluster'][node_number | int]['mgmt_interface'] }}   # The management interface name
-      method: dhcp
-    bond0:
-      interfaces:
-      - name: {{ settings['harvester_network_config']['cluster'][node_number | int]['vagrant_interface'] }}
-      method: dhcp
+  management_interface:
+    interfaces:
+    - name: {{ settings['harvester_network_config']['cluster'][0]['mgmt_interface'] }}  # The management interface name
```
Comment (bk201) [inline vagrant-pxe-harvester/ansible/roles/harvester/templates/config-join.yaml.j2:20]:
```suggestion
    - name: {{ settings['harvester_network_config']['cluster'][node_number | int]['mgmt_interface'] }}  # The management interface name
```

File: vagrant-pxe-harvester/ansible/roles/harvester/templates/config-join.yaml.j2 [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # example from https://github.com/harvester/ipxe-examples/blob/main/general/config-join.yaml
 
+scheme_version: 1
```
Comment (bk201) [inline vagrant-pxe-harvester/ansible/roles/harvester/templates/config-join.yaml.j2:3]: We need to add these to `general/` examples as well.

**Comment by bk201** [comment]
We also need to modify the example in the [Equinix path](https://github.com/harvester/ipxe-examples/tree/main/equinix). Including join-node configuration and the check if those kernel parameter hacks still works: `harvester.install.networks.harvester-mgmt.method=dhcp harvester.install.networks.harvester-mgmt.bond_options.mode=balance-tlb `.

File: vagrant-pxe-harvester/ansible/roles/harvester/templates/config-join.yaml.j2 (hunk 2 above)
Comment (tjjh89017) [reply vagrant-pxe-harvester/ansible/roles/harvester/templates/config-join.yaml.j2:3]: done

**Review by bk201 — APPROVED** [review]

**Comment by bk201** [comment]
Create `v1.0` branch from `main` branch for v1.0.x configuration files.
The main branch is now for `v1.1.0`.

**Pull request merged**